src/routes/e2e/widget/modal/+page.svelte

```diff
@@ -2,58 +2,37 @@
   import { onMount } from 'svelte';
   import { page } from '$app/stores';
 
-  import Widget, { configuration, modal, journey, user } from '$package/modal';
+  import Widget, { configuration, component, journey, user } from '$package/index';
+
+  const config = configuration();
+  const componentEvents = component();
+  const journeyEvents = journey();
 
   let authIndexValue = $page.url.searchParams.get('authIndexValue');
   let journeyParam = $page.url.searchParams.get('journey');
   let suspendedIdParam = $page.url.searchParams.get('suspendedId');
 
-  // TODO: Use a more specific type
   let userResponse: any | null;
-  let widget: Widget;
   let widgetEl: HTMLDivElement;
 
   async function logout() {
     await user.logout();
     userResponse = null;
   }
 
-  // TODO: Investigate why the parameter types are needed here
-  modal.onMount((dialog: HTMLDialogElement, form: HTMLFormElement) => {
-    console.log(dialog);
-    console.log(form);
-
-    // Calling this on mount is not good if using HMR as it results in a ton of requests on change
-    // journey.start();
+  componentEvents.subscribe((event) => {
```

Comment:
We can now `subscribe` to the `componentEvents`, or whatever you call it, and it can be done anywhere. If you subscribe after the Widget has been initialized, it will just return the current event's object which will indicate if the modal is already mounted or not.